wingpanel-interface/FocusManager.vala

```diff
@@ -128,6 +117,21 @@ public class WingpanelInterface.FocusManager : Object {
                 }
 
                 if (event.get_type () == MOTION || event.get_type () == TOUCH_UPDATE) {
+                    if (!started_drag) {
+                        started_drag = true;
+                        event.get_coords (out start_x, out start_y);
+
+                        return Clutter.EVENT_PROPAGATE;
+                    }
+
+                    float current_x, current_y;
+                    event.get_coords (out current_x, out current_y);
+
+                    var drag_threshold = Clutter.Settings.get_default ().dnd_drag_threshold;
+                    if ((start_x - current_x).abs () <= drag_threshold && (start_y - current_y).abs () <= drag_threshold) {
```

Comment:
Should this be `&&` or `||` ? I would imagine you want `||` for drags that are mostly along an axis, no?